docker/testsysds.Dockerfile

```diff
@@ -78,6 +78,13 @@ RUN apt-get install -y --no-install-recommends \
 	&& rm -rf installDependencies.R \
 	&& rm -rf /var/lib/apt/lists/*
 
+RUN apt-get install -y --no-install-recommends \
+	python3 python3-pip && \
+	apt-get clean && \
+	python3 -m pip install --upgrade pip && \
+	pip3 install numpy scipy && \
+    echo "Python3 installed"
```

Comment:
This is fine to add, however, i would prefer not to.

We have an---unrelated to your PR---problem with the testsysds docker file that would require a larger fix to make it build properly. Therefore, it will take some time before we would run the update and therefore be able to enable the test to call python from inside the system.

Fundamentally this blocks your PR, because your work relies on Python being installed on the machine you want to test on.

Reply:
Since I’m not familiar with the SystemDS Python API, I’ll study it and proceed as you suggested.

Reply:
@min-guk Hi, I have worked recently on the python api, if I can help you lmk

Reply:
Thank you both for your kind words. I would like to ask a question to confirm one thing.

Should I implement a Python API to make the roll function, which is implemented in Java, callable from Python, and then write Python test code to compare it with NumPy?

And for such examples, should I refer to https://github.com/apache/systemds/pull/2085 and https://github.com/apache/systemds/pull/2097?

Reply:
jep, in the Python test you can compare to numpy, you can take a look at the test in #2085 . (#2097 would be an example for an operator with two matrices as input, but since you got one matrix sa input #2085 fits better), 
so you can extend the https://github.com/apache/systemds/blob/main/src/main/python/systemds/operator/nodes/matrix.py with your operator.
You can also take a look at the quantile operator there on how to pass scalars to your function call (for your shift paramerter)

Reply:
Thank you for the clarification and the references. I'll take a look and proceed accordingly.

Reply:
@e-strauss Thanks to your help, I was able to successfully implement and complete the Python API.

If possible, could you share an example of a Python API that calls a Spark function, along with its corresponding test code? It would be really helpful to me.

Reply:
We currently do not actively assign spark instructions through the python API, because we did not expose the flag for it yet. 

The CP test would be sufficient from python and then internally you can set the function test to test the operation with SP enabled for an input that you know the output for. 

Together this would be sufficient.

Reply:
Thank you for letting me know. I will verify CP by comparing it with numpy in the Python test, and for SP, I will verify it by comparing the output with CP using the Java test.